### PR TITLE
Batons passive drain a small amount of charge if left on

### PR DESCRIPTION
**Pull request by Bm0n**
## What Does This PR Do
Makes it so that batons drain 100 units of charge per two seconds

That's about a full baton hit every 20 seconds it is left on doing nothing.

## Why It's Good For The Game

Meant to work with PR https://github.com/ParadiseSS13/Paradise/pull/19957

The Baton is still a very powerful item even after the rework and this at least makes there have to be a deliberate action to use it kind of like the telebaton.

I think this is a needed change for my aforementioned PR but could also work by itself.

Really want to hear what you all think about the drain rate, I think a 100 per two seconds is a good number but am open to change it depending on feedback.

## Testing

In-game

## Changelog
:cl: Bmon
add: The baton will now slowly run out of charge if left on.
/:cl:

### CODE REVIEW

**Comment by matttheficus** [comment]
How long does it last with a Bluespace Battery and/or does a Yellow Core make it infinite?

**Comment by Bm0n** [comment]
> How long does it last with a Bluespace Battery and/or does a Yellow Core make it infinite?

Yellow cores are gone, old man! But blue spaces are 13.32 minutes for it to fully deplete

**Comment by matttheficus** [comment]
I forgor 💀 This is why Science is a mistake.

**Comment by Bm0n** [comment]
> I forgor 💀 This is why Science is a mistake.

I LIED TO YOU THEY STILL EXIST!  Uh, I think it regenerates power fast enough for it not to be an issue... at least they're locked behind some dumb xenobio slime requirement

should really rework them

**Comment by Scribble-Sheep** [comment]
Interesting. So, if this is meant as a way to prevent macro-users from being able to do the instant baton thing that Sean mentioned, I don't think it'd be so? It'd just add an extra button press to the macro, unless there's a delay to turning the baton on/off that I'm not aware of.

That said, I think this kinda stands on its own as a baton balance pr. Whether or not batons are in need of that isn't something I can attest to, since I have next to no experience with them on either end.

**Comment by procdrone** [comment]
as a player that enjoys slower combat and less reliance on more and more hotkeys, I don't like this.

**Comment by Bm0n** [comment]
> as a player that enjoys slower combat and less reliance on more and more hotkeys, I don't like this.

good news, this would make things slower.

could always click your baton if you're really hellbent against using the Z key

**Comment by Bm0n** [comment]
Yeah even though drawing is gone I still think batons are stupidly overpowered as is so am leaving this up.

**Comment by Rurik123** [comment]
Nice little warning bzzt audio cue before you're smacked with a baton. I like it.

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Comment by procdrone** [comment]
i know it doesnt matter but im withdrawing my "i dont like it" to "i dont care either way", I can see some arguments for it.

**Review by S34NW — APPROVED** [review]

File: code/game/objects/items/weapons/stunbaton.dm [hunk 1]
```diff
@@ -35,6 +35,7 @@
 
 /obj/item/melee/baton/Destroy()
 	if(cell?.loc == src)
+		STOP_PROCESSING(SSobj, src)
```
Comment (SteelSlayer) [inline code/game/objects/items/weapons/stunbaton.dm:38]: I think this should go outside of any conditional, just to be safe. Batons should call stop processing when they're destroyed no matter where the cell is.

**Review by SteelSlayer — CHANGES_REQUESTED** [review]
You also need to add a `STOP_PROCESSING` to `screwdriver_act` because it's possible the user can eject the cell while the baton is on, thus creating runtimes in `process()`.

**Review by SteelSlayer — APPROVED** [review]

File: code/game/objects/items/weapons/stunbaton.dm [hunk 2]
```diff
@@ -132,22 +134,35 @@
...
 	update_icon()
 	add_fingerprint(user)
 
+/obj/item/melee/baton/process()
+	cell.use(100)
+	if(!cell.charge)
+		turned_on = FALSE
+		playsound(src, "sparks", 75, TRUE, -1)
+		update_icon()
+		STOP_PROCESSING(SSobj, src)
```
Comment (farie82) [inline code/game/objects/items/weapons/stunbaton.dm:165]: Basically having these 4 lines copy pasted over the file hurts my eyes. Can you make a `turn_on` and `turn_off` proc?

Comment (Bm0n) [reply code/game/objects/items/weapons/stunbaton.dm:165]: i could try this but also not too confident in coding abilities

anyways it'll be at least week until i can, going to be busy for a bit

**Comment by matttheficus** [comment]
Thinking more on this, if the goal for this PR is to prevent people from `Drawing Baton > Instantly ready to hit people`, wouldn't simply preventing your from putting it away unless it was off be a better method of doing this (much like welders)? 

The general worry from some on the Balance Team is passive drain either being negligible via upgrades OR it losing charge passively over a long-standing fight (i.e. Nukies, Armored Clings, etc).

**Comment by Bm0n** [comment]
> Thinking more on this, if the goal for this PR is to prevent people from `Drawing Baton > Instantly ready to hit people`, wouldn't simply preventing your from putting it away unless it was off be a better method of doing this (much like welders)?

well you see if you do that then it'd be much more clunky to put your baton away in an emergency let's say after you yourself get hit by a baton. also, how would that make much sense?

might I add that the welder feature you're talking about is new and very jank, you can still pocket lit welders for instance

> The general worry from some on the Balance Team is passive drain either being negligible via upgrades OR it losing charge passively over a long-standing fight (i.e. Nukies, Armored Clings, etc).

It can't be both.

**Comment by zzzmike** [comment]
This seems to be newbie-unfriendly. If the goal is to nerf sec / buff antags (like most balance changes seem to be recently), I think it's more friendly to newbies if you just make each hit drain more charge.

**Comment by Bm0n** [comment]
> This seems to be newbie-unfriendly. If the goal is to nerf sec / buff antags (like most balance changes seem to be recently), I think it's more friendly to newbies if you just make each hit drain more charge.

You've mostly missed the point of this PR, it isn't to nerf batons themselves but to stop people from being immediately able to draw and use a baton with no warning.

Regardless, I don't think batons should be newbie-friendly in the first place. If you're getting yourself into combat or sec you should at least be an intermediate level player

**Comment by zzzmike** [comment]
> You've mostly missed the point of this PR, it isn't to nerf batons themselves but to stop people from being immediately able to draw and use a baton with no warning. A tactic I most commonly see used by both antags and sec.
 
 Ok, fair enough. I don't think it really makes a noticeable difference in that goal, as a veteran player will just hit the keybind a split second before clicking which doesn't really provide meaningful warning. For a newbie player, sure. But as I said I don't think it's a good idea for a change which is newbie-unfriendly.

> Regardless, I don't think batons should be newbie-friendly in the first place. If you're getting yourself into combat or sec you should at least be an intermediate level player

I can't agree with you on this - everyone is gonna be a newbie sec (or newbie antag, if you really wanna make the argument that a stun baton nerf is aimed equally at antags...) at one point - everyone starts somewhere. (regardless of the minor hour requirement which you can bypass by mostly AFKing if you want). A change that doesn't really have a noticeable effect when veterans baton people but can be frustrating for newbies isn't a good one in my eyes. But hey, since it's been testmerged this long it's probably gonna be fullmerged regardless.

**Comment by Kuzo1971** [comment]
So with this PR, I can't help but get the vibe it's change for the sake of change.

I think the largest issue we agree is surprise batonning while it's on currently in a bag, and I believe an alternative PR is already up to ensure that it can't be put in a bag while on.

A lot of the benefits this PR has are niche at best, while the cons apply in almost every scenario. While yes, it's not impossible to regulate combat by turning it on and off, we're punishing players for not having even more micro managment for an issue I don't believe has ever been called an issue before. 

This PR, respectfully fails your stated objective of inventory/baton relations and has far more implications outside of that, whether intended or not.

**Comment by OctusGit** [comment]
This doesn't stop people from instantly batoning you with a quickdraw. Adding 1 key input doesn't change the gameplay or solve the issue aside from being what some have said a very anti quality of life change. This PR is said "not" to be a nerf to stun batons but in the same PR description you says its "still a powerful weapon" which sorta feels disingenuous. 

This is a nerf to stun batons, and its a very odd nerf that ingame I've yet to see solve the main issue. I consistently see it more just inconveniencing people and making new sec players look more bald. I still constantly see quick batons hitting people in the face in less than a second. 

I dont think the baton has any issues aside from being strong but that's kind of the balance for security as melee is dangerous versus alot of antags who have much better melee in their kits. Vamps, Tots, Etc

**Comment by Bm0n** [comment]
> This doesn't stop people from instantly batoning you with a quickdraw. Adding 1 key input doesn't change the gameplay or solve the issue aside from being what some have said a very anti quality of life change. This PR is said "not" to be a nerf to stun batons but in the same PR description you says its "still a powerful weapon" which sorta feels disingenuous.
> 
> This is a nerf to stun batons, and its a very odd nerf that ingame I've yet to see solve the main issue. I consistently see it more just inconveniencing people and making new sec players look more bald. I still constantly see quick batons hitting people in the face in less than a second.
> 
> I dont think the baton has any issues aside from being strong but that's kind of the balance for security as melee is dangerous versus alot of antags who have much better melee in their kits. Vamps, Tots, Etc

You fail to see the context and the why of this PR being made. It was to add a delay in drawing and using a baton from when I trying to port hotkeys. That didn't work out but I decided to leave this PR up anyways because yeah batons are powerful. This PR doesn't change how batons actually physically work but does stop people from leaving them on at all times and adds an action in between pulling it out and using it.

In my opinion, the baton shouldn't get quality-of-life features as it is a very powerful weapon that is desperately in need of a rebalance. Making it more clunky on purpose is a good thing.

People will probably not read the changelog and leave their batons on only to realise that it's drained when they go to use it but that can't be avoided, it'll happen once to someone and they probably won't be repeated again.

Making it so that you have to push a single button before using it is not much and I am slightly taken off by the large response this PR is getting. Oh well, that's what I get for dipping my toes into balance.

**Comment by Bm0n** [comment]
> So with this PR, I can't help but get the vibe it's change for the sake of change.
>
> A lot of the benefits this PR has are niche at best, while the cons apply in almost every scenario. While yes, it's not impossible to regulate combat by turning it on and off, we're punishing players for not having even more micro managment for an issue I don't believe has ever been called an issue before.
> 
> This PR, respectfully fails your stated objective of inventory/baton relations and has far more implications outside of that, whether intended or not.

How so?

**Comment by Kuzo1971** [comment]
> > So with this PR, I can't help but get the vibe it's change for the sake of change.
> > A lot of the benefits this PR has are niche at best, while the cons apply in almost every scenario. While yes, it's not impossible to regulate combat by turning it on and off, we're punishing players for not having even more micro managment for an issue I don't believe has ever been called an issue before.
> > This PR, respectfully fails your stated objective of inventory/baton relations and has far more implications outside of that, whether intended or not.
> 
> How so?

So, your overall aim was to provide a barrier to prevent batons instantly being drawn. You attempted this in the most indirect way possible, which has affects far broader than what your goal was. 

Firstly, in regard to your objective, you mitigate instant baton draws by a negligble amount. This is one of those niche benefits. However, now I'm punished for a neglible benefit if I put it away during a chase in the heat of the fight. You could argue "well, adapt to it" to which I ask why? Why hinder myself, or new officers for an objective that hasn't been met. To knowledge(I'm not entirely sure, but because I was busy beating a cling over the head) it ALSO drains while being used in combat, which adds another level and burden. Both of these negatives are WAY broader than the scope you intended for this PR when there's a solution that avoids both of these negatives by making something that can't be stored while on.

If you want to make balance, combat related changes, this seems more like a PR on the mark. I disagree with even those changes, but I'm sticking to what you aimed for in the post.

**Comment by Bm0n** [comment]
> You could argue "well, adapt to it" to which I ask why?

Because it's one of the most potent melee weapons anyone can get their hands on right now. I never said this wasn't a balance change, just that this doesn't directly nerf batons in how they function. Of course, this PR is a balance change, it was made to slow down people being able to use them which it does but it also adds clunk to the baton purposely because it's super powerful.

I don't know how you missed that, I mean this PR is even labelled balance.

**Comment by Kuzo1971** [comment]
> > You could argue "well, adapt to it" to which I ask why?
> 
> Because it's one of the most potent melee weapons anyone can get their hands on right now. I never said this wasn't a balance change, just that this doesn't directly nerf batons in how they function. Of course, this PR is a balance change, it was made to slow down people being able to use them which it does but it also adds clunk to the baton purposely because it's super powerful.
> 
> I don't know how you missed that, I mean this PR is even labelled balance.

I mean in relation to combat, specifically. Sorry that was unclear, I think everything I said still stands despite that. I'm not going to copy and paste my previous comment, but yeah, still on the lines of that. If I wanted it to be a more "deliberate" action, I'd have gone with the off-in-bag solution. Like I said, and what's in your "why it's good for the game section" has implications way more than your stated intention while not exactly meeting what your aim is, In my opinion, 

This PR doesn't make batons more deliberate, it makes them more tedious, hence what I pointed out in the neglible difference in speed drawing them pre test merge and now.

**Comment by Bm0n** [comment]
>I mean in relation to combat, specifically. Sorry that was unclear, I think everything I said still stands despite that. I'm not going to copy and paste my previous comment, but yeah, still on the lines of that. If I wanted it to be a more "deliberate" action, I'd have gone with the off-in-bag solution. Like I said, and what's in your "why it's good for the game section" has implications way more than your stated intention while not exactly meeting what your aim is, In my opinion,

Even in relation to combat, it is still a very powerful weapon that doesn't deserve to be as powerful as it currently is, especially when compared to all the other stun weapons it is the best hands down.

I stated it very clearly in the "Why It's Good For The Game" section that I did this because the baton is a powerful weapon. While this original was tied to another PR I still think the baton is a powerful weapon even without that PR.

>This PR doesn't make batons more deliberate, it makes them more tedious, hence what I pointed out in the neglible difference in speed drawing them pre test merge and now.

A tedious action is a deliberate action because you have to do that action to use your baton, the point of this PR. We should not be optimising or making powerful weapons not tedious. From what I see in game (even though my time is very limited right now) and hear elsewhere this is slowing people down in using their batons which was the aim of this PR, if it didn't no one would be complaining about it right now.

**Comment by hal9000PR** [comment]
despite liking the intent (forcing people to turn batons on when they draw them)
I am not entirely sold on the execution. reducing securities power in long drawn out fights is not ideal especially with the push to make combat more drawn out. 
3 balance team objections, filing as closed.

**Pull request closed without merging**